### PR TITLE
fix(sms): Use a distinct phone number for every SMS test.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
testPhoneNumber and testPhoneNumberCountry are no longer needed.

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - r?

This PR is because @jrgm discovered that tests running in parallel would clobber each other because they all used the same phone number. I am attempting to fix that by using different phone numbers for every test, but since there are only 6 digits that change, I still delete SMSs before every test.

File: tests/functional/fx_desktop_handshake.js [hunk 1]
```diff
@@ -144,46 +141,45 @@ define([
...
-            webChannelResponses: {
-              'fxaccounts:fxa_status': {
-                signedInUser: browserSignedInAccount
-              }
+      const testPhoneNumber = TestHelpers.createPhoneNumber();
+      let signinUrlWithSigninCode;
+
+      return this.remote
+        // The phoneNumber is reused across tests, delete all
+        // if its SMS messages to ensure a clean slate.
```
Comment (philbooth) [inline tests/functional/fx_desktop_handshake.js]: nit: s/if/of/

**Comment by philbooth** [comment]
LGTM! r+

File: tests/lib/helpers.js [hunk 2]
```diff
@@ -30,15 +31,34 @@ define([
     return template.replace('{id}', Math.random()) + '@restmail.net';
   }
 
+  function createPhoneNumber () {
+    // start with an area code and known good first number,
+    // append a 6 character suffix.
+    const prefix = '9164';
+    // For a total of 10 characters, we need
+    let suffix = Math.floor(Math.random() * Math.pow(10, 6));
+    suffix = padright(suffix, 6, '0');
```
Comment (shane-tomlinson) [inline tests/lib/helpers.js]: I'm not sure whether padleft or padright is better here, or whether it makes any difference at all.

File: tests/lib/helpers.js [hunk 3]
```diff
@@ -30,15 +31,34 @@ define([
     return template.replace('{id}', Math.random()) + '@restmail.net';
   }
 
+  function createPhoneNumber () {
+    // start with an area code and known good first number,
+    // append a 6 character suffix.
+    const prefix = '9164';
+    // For a total of 10 characters, we need
```
Comment (shane-tomlinson) [inline tests/lib/helpers.js]: This comment is incomplete.

File: tests/lib/helpers.js [hunk 4]
```diff
@@ -30,15 +31,34 @@ define([
...
+    let suffix = Math.floor(Math.random() * Math.pow(10, 6));
```
Comment (shane-tomlinson) [inline tests/lib/helpers.js]: Extract the 6 to a constant, or calculate it.

File: tests/lib/helpers.js [hunk 5]
```diff
@@ -30,15 +31,34 @@ define([
     return template.replace('{id}', Math.random()) + '@restmail.net';
   }
 
+  function createPhoneNumber () {
+    // start with an area code and known good first number,
+    // append a 6 character suffix.
+    const prefix = '9164';
```
Comment (shane-tomlinson) [inline tests/lib/helpers.js]: Maybe pass in the prefix and desired length, setting defaults for each.

**Pull request merged**